netty/src/main/java/io/grpc/netty/NettyClientHandler.java

```diff
@@ -244,6 +252,9 @@ void returnProcessedBytes(Http2Stream stream, int bytes) {
   private void onHeadersRead(int streamId, Http2Headers headers, boolean endStream) {
     NettyClientStream.TransportState stream = clientStream(requireHttp2Stream(streamId));
     stream.transportHeadersReceived(headers, endStream);
+    if (keepAliveManager != null) {
+      keepAliveManager.onDataReceived();
```

Comment:
We'll also want to trigger this on `FrameListener.onPingAckRead` and `FrameListener.onPingRead` (it would be a new override).